The MSSQL/DB2 devservices can't find container-license-acceptance.txt

### Describe the bug

When running the devmode with `quarkus-jdbc-mssql` it throw error `The image mcr.microsoft.com/mssql/server:2022-latest requires you to accept a license agreement. Please place a file at the root of the classpath named container-license-acceptance.txt, e.g. at src/test/resources/container-license-acceptance.txt`

This happening with 999-SNAPSHOT only and it's probably caused by https://github.com/quarkusio/quarkus/pull/52413 (didn't check it)

### Expected behavior

Not throw the error when the `src/test/resources/container-license-acceptance.txt` is present in project and contain the correct line.

### Actual behavior

Throwing the 
```java
ERROR [io.quarkus.deployment.dev.IsolatedDevModeMain] (main) Failed to start quarkus: java.util.concurrent.CompletionException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mcr.microsoft.com/mssql/server:2022-latest
        at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(CompletableFuture.java:323)
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:359)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:364)
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1828)                                                                                                                                                                                                                                                                         
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1817)                                                                                                                                                                                                                                                                        
        at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:511)                                                                                                                                                                                                                                                                                          
        at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1450)                                                                                                                                                                                                                                                                         
        at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2019)                                                                                                                                                                                                                                                                                      
        at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)                                                                                                                                                                                                                                                                             
Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image mcr.microsoft.com/mssql/server:2022-latest                                                                                                                                                                                                                              
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:346)                                                                                                                                                                                                                                                                                  
        at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:317)                                                                                                                                                                                                                                                                                    
        at io.quarkus.deployment.builditem.DevServicesRegistryBuildItem.reallyStart(DevServicesRegistryBuildItem.java:179)                                                                                                                                                                                                                                                    
        at io.quarkus.deployment.builditem.DevServicesRegistryBuildItem.start(DevServicesRegistryBuildItem.java:134)                                                                                                                                                                                                                                                          
        at io.quarkus.deployment.builditem.DevServicesRegistryBuildItem.lambda$startSelectedServices$2(DevServicesRegistryBuildItem.java:118)                                                                                                                                                                                                                                 
        at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1825)                                                                                                                                                                                                                                                                         
        ... 5 more                                                                                                                                                                                                                                                                                                                                                            
Caused by: java.lang.IllegalStateException: The image mcr.microsoft.com/mssql/server:2022-latest requires you to accept a license agreement. Please place a file at the root of the classpath named container-license-acceptance.txt, e.g. at src/test/resources/container-license-acceptance.txt. This file should contain the line:                                         
  mcr.microsoft.com/mssql/server:2022-latest                                                                                                                                                                                                                                                                                                                                  
        at org.testcontainers.utility.LicenseAcceptance.assertLicenseAccepted(LicenseAcceptance.java:30)                                                                                                                                                                                                                                                                      
        at org.testcontainers.mssqlserver.MSSQLServerContainer.configure(MSSQLServerContainer.java:67)                                                                                                                                                                                                                                                                        
        at io.quarkus.devservices.mssql.deployment.MSSQLDevServicesProcessor$QuarkusMSSQLServerContainer.configure(MSSQLDevServicesProcessor.java:118)                                                                                                                                                                                                                        
        at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:326)                                                                                                                                                                                                                                                                                  
        ... 10 mor
```

### How to Reproduce?

1. `git clone -b mssql-dev-services https://github.com/jedla97/quarkus-reproducers.git`
2. `cd quarkus-reproducers`
3. `quarkus dev`
4. Devmode fail to start with error above

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

25.0.2-tem, 21.0.9-tem

### Quarkus version or git rev

main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/devservices/db2/src/main/java/io/quarkus/devservices/db2/deployment/DB2DevServicesProcessor.java
+++ b/extensions/devservices/db2/src/main/java/io/quarkus/devservices/db2/deployment/DB2DevServicesProcessor.java
@@ -153,6 +153,7 @@ public class DB2DevServicesProcessor {
         private final boolean podman;
 
         private final String hostName;
+        private ClassLoader creationClassloader;
 
         public QuarkusDb2Container(Optional<String> imageName, OptionalInt fixedExposedPort,
                 String defaultNetworkId, boolean useSharedNetwork, boolean podman) {
@@ -162,6 +163,23 @@ public class DB2DevServicesProcessor {
             this.useSharedNetwork = useSharedNetwork;
             this.podman = podman;
             this.hostName = ConfigureUtil.configureNetwork(this, defaultNetworkId, useSharedNetwork, "db2");
+            creationClassloader = Thread.currentThread().getContextClassLoader();
+        }
+
+        @Override
+        public void start() {
+            // Ideally StartupActionImpl would set the deployment classloader as the TCCL, but it sets the augmentation classloader as the TCCL
+            // In normal mode this doesn't matter because the augmentation classloader can see application classes, but in continuous testing/dev mode, it can't
+            // As a tactical workaround, stash away the deployment classloader and use it for the start, so that test containers classes can find application resources
+            ClassLoader orig = Thread.currentThread().getContextClassLoader();
+            try {
+                Thread.currentThread().setContextClassLoader(creationClassloader);
+                super.start();
+            } finally {
+                Thread.currentThread().setContextClassLoader(orig);
+                // Assume the same container instance won't start() twice, and null out the CL to be cautious about CL leaks
+                creationClassloader = null;
+            }
         }
 
         @Override

--- a/extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java
+++ b/extensions/devservices/mssql/src/main/java/io/quarkus/devservices/mssql/deployment/MSSQLDevServicesProcessor.java
@@ -102,6 +102,7 @@ public class MSSQLDevServicesProcessor {
         private final boolean useSharedNetwork;
 
         private final String hostName;
+        private ClassLoader creationClassloader;
 
         public QuarkusMSSQLServerContainer(Optional<String> imageName, OptionalInt fixedExposedPort,
                 String defaultNetworkId, boolean useSharedNetwork) {
@@ -111,6 +112,23 @@ public class MSSQLDevServicesProcessor {
             this.fixedExposedPort = fixedExposedPort;
             this.useSharedNetwork = useSharedNetwork;
             this.hostName = ConfigureUtil.configureNetwork(this, defaultNetworkId, useSharedNetwork, "mssql");
+            creationClassloader = Thread.currentThread().getContextClassLoader();
+        }
+
+        @Override
+        public void start() {
+            // Ideally StartupActionImpl would set the deployment classloader as the TCCL, but it sets the augmentation classloader as the TCCL
+            // In normal mode this doesn't matter because the augmentation classloader can see application classes, but in continuous testing/dev mode, it can't
+            // As a tactical workaround, stash away the deployment classloader and use it for the start, so that test containers classes can find application resources
+            ClassLoader orig = Thread.currentThread().getContextClassLoader();
+            try {
+                Thread.currentThread().setContextClassLoader(creationClassloader);
+                super.start();
+            } finally {
+                Thread.currentThread().setContextClassLoader(orig);
+                // Assume the same container instance won't start() twice, and null out the CL to be cautious about CL leaks
+                creationClassloader = null;
+            }
         }
 
         @Override

